koku/api/report/queries.py

```diff
@@ -1202,6 +1202,10 @@ def _ranked_list(self, data_list, ranks, rank_fields=None):  # noqa C901
             account_aliases = account_aliases.drop_duplicates(subset="account")
         data_frame.drop(columns=drop_columns, inplace=True, errors="ignore")
         data_frame = data_frame.merge(ranks_by_day, how="right", on=merge_on)
+        data_frame["subscription_name"] = data_frame.apply(
+            lambda row: row["subscription_guid"] if pd.isna(row["subscription_name"]) else row["subscription_name"],
+            axis=1,
+        )
```

Comment:
This is `subscription_name/subscription_guid` Azure specific, but we are in a base class here. So I think we need to have a `if self.is_azure`, similar to the way we have `self.is_aws` just a couple lines above here. If we don't have a `is_azure` property, then we will need to add one.